EIPS/eip-2612.md

```diff
@@ -169,25 +169,29 @@ ERC-712 typed messages are included because of its wide spread adoption in many
 
 
 ## Backwards Compatibility
-There are currently two slightly differing implementations of this ERC, and we are forced to make a choice here for specificity.
-This implies that the given ERC deviates from the implementation given in the Dai and Chai ERC-20 contracts. There, the `permit` method takes `nonce` as an additional argument, and the `uint256 value` argument is exchanged for `bool approval`, admitting binary approvals only. There is also a slight difference in argument names. The specification presented here is in line with the implementation in [Uniswap-v2](https://github.com/uniswap/uniswap-v2-core). This mismatch is a little unfortunate, but not very different from the variations found in ERC-20 contracts. 
+There are already a couple of `permit` functions in token contracts implemented in contracts in the wild, most notably the one introduced in the `dai.sol`.
+
+Its implementation differs slightly from the presentation here in that:
+- instead of taking a `value` argument, it takes a bool `allowed`, setting approval to 0 or `uint(-1)`.
+- the `deadline` argument is instead called `expiry`. This is not just a syntactic change, as it effects the contents of the signed message.
+
+There is also an implementation in the token [`Stake`](https://etherscan.io/address/0x0Ae055097C6d159879521C384F1D2123D1f195e6#code) with the same ABI as `dai` but with different semantics: it lets users issue "expiring approvals", that only allow `transferFrom` to occur while `expiry >= block.timestamp`.
+
+The specification presented here is in line with the implementation in [Uniswap-v2](https://github.com/uniswap/uniswap-v2-core).
 
 ## Test Cases
-Some basic test vector can be found here https://github.com/Uniswap/uniswap-v2-core/blob/master/test/UniswapV2ERC20.spec.ts.
-Additional test vectors for the Dai/Chai-like implementation are given at [chai.t.sol](https://github.com/dapphub/chai/blob/master/src/test/chai.t.sol).
 
+Some basic tests can be found here https://github.com/Uniswap/uniswap-v2-core/blob/master/test/UniswapV2ERC20.spec.ts.
 
 ## Implementation
 [UniswapV2ERC20.sol](https://github.com/Uniswap/uniswap-v2-core/blob/master/contracts/UniswapV2ERC20.sol)
-[Dai.sol](https://github.com/makerdao/dss/blob/master/src/dai.sol)
-[Chai.sol](https://github.com/makerdao/dss/blob/master/src/dai.sol)
-
-Note that the latter two implementations differ slightly from the presentation given here.
 
 ## Security Considerations
 
 Though the signer of a `Permit` may have a certain party in mind to submit their transaction, another party can always front run this transaction and call `permit` before the intended party. The end result is the same for the `Permit` signer, however.
 
+Since the ECREC precompile fails silently and just returns the zero address as `signer` when given malformed messages, it is important to ensure `owner =/= address(0)` to avoid `permit` from creating an approval to spend "zombie funds" belong to the zero address.
```

Comment:
I think `ecrecover precmopile` is a more commonly used term.

Also could refer to https://github.com/ethereum/EIPs/pull/2602 here.